pkgs/os-specific/linux/i2c-tools/default.nix

```diff
@@ -1,12 +1,18 @@
-{ lib, stdenv, fetchurl, perl, read-edid }:
+{ lib
+, stdenv
+, fetchgit
+, perl
+, read-edid
+}:
 
 stdenv.mkDerivation rec {
   pname = "i2c-tools";
   version = "4.2";
 
-  src = fetchurl {
-    url = "https://www.kernel.org/pub/software/utils/i2c-tools/${pname}-${version}.tar.xz";
-    sha256 = "1mmc1n8awl3winyrp1rcxg94vjsx9dc1y7gj7y88blc2f2ydmwip";
+  src = fetchgit {
+    url = "https://git.kernel.org/pub/scm/utils/i2c-tools/i2c-tools.git";
```

Comment:
is there an advantage to checking out from git? Generally it's less preferred to use git as you have to download the entirety of the git history, then throw it all away.

Reply:
ease of using an arbitrary commit

doesn't it do a shallow clone?

Reply:
Hmm, it does `clean_git fetch ${builder:+--progress} --depth 1 origin +"$ref" || return 1`

Do you feel like it's likely that using arbitrary commits will be needed?

Reply:
i tried an arbitrary commit because i found it easier to blame the source than to go about packaging properly
hardly a good reason :P

though i also have the impression it's easier to switch to another source like this
(no need to change the rev and hash)

i seem to have a general preference for git sources, but give the word and i'll switch it back

Reply:
I guess I don't feel strong about it :)